internal/streams/streams.go

```diff
@@ -50,13 +50,8 @@ func streamFromBucket(bucket, streamInBucket int) int {
 }
 
 func (s *IDGenerator) GetStream() (int, bool) {
-	// based closely on the java-driver stream ID generator
-	// avoid false sharing subsequent requests.
-	offset := atomic.LoadUint32(&s.offset)
-	for !atomic.CompareAndSwapUint32(&s.offset, offset, (offset+1)%s.numBuckets) {
-		offset = atomic.LoadUint32(&s.offset)
-	}
-	offset = (offset + 1) % s.numBuckets
+	// Reduce collisions by offsetting the starting point
+	offset := atomic.AddUint32(&s.offset, 1)
```

Comment:
why without the modulo?

Reply:
I am sorry, I can't grasp it, could you give more info ?

Reply:
I am trying to understand why in the new version we do not have `% s.numBuckets` part

Reply:
because it is done further down the code

Reply:
ok, I missed it